processor/groupbytraceprocessor/event.go

```diff
@@ -254,7 +254,7 @@ func (em *eventMachine) handleEventWithObservability(event string, do func() err
 // doWithTimeout wraps a function in a timeout, returning whether it succeeded before timing out.
 // If the function returns an error within the timeout, it's considered as succeeded and the error will be returned back to the caller.
 func doWithTimeout(timeout time.Duration, do func() error) (bool, error) {
-	done := make(chan error)
+	done := make(chan error, 1)
```

Comment:
I know that it fixes the problem, but I don't understand why. Do you have any docs that explains this?

Reply:
Actually it's simple and explained in https://tour.golang.org/concurrency/3. I can try to clarify the difference

If we have an unbuffered channel, then send operation blocks until someone reads from the channel. But no one does in case of timeout. Thus, the send operation in goroutine blocks forever.

Buffered channel allows to send items in non-blocking mode if the buffer isn't full (it isn't in this case as long as there is only 1 send). Finally, even if timeout is reached, the error will be sent into the channel, allowing go function to finish and the channel is destroyed due to zero reference count

Reply:
That's wonderful, thanks for the lesson! I'm sure I've seen that page quite a few times already, but never had this insight.